### PR TITLE
Fix: Correct the navigation to anchor

### PR DESCRIPTION
**Pull request by Dedelweiss**
The browser compatibility of [`location.hash`](https://developer.mozilla.org/en-US/docs/Web/API/Location/hash) seems to be out of date. That's why in this fix, I suggest using [`history.replaceState`](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState).

## Before:

https://github.com/lampepfl/dotty/assets/44496264/a1399b0b-e4e8-4754-b146-661b59a01824

## After:

https://github.com/lampepfl/dotty/assets/44496264/22f38852-0cc5-48eb-a6e7-699f11628401

Fixes: #17182

### CODE REVIEW

**Comment by Dedelweiss** [comment]
Another solution would be to add a setTimeout() with the window.location.hash = "";

File: scaladoc/resources/dotty_res/scripts/ux.js [hunk 1]
```diff
@@ -285,7 +285,7 @@ document
         getTocListElement(id).parentElement.classList.toggle("active");
       }
       if (lastEntry.isIntersecting) {
-        window.location.hash = "";
+        history.replaceState(null, "", window.location.pathname + window.location.search);
```
Comment (Florian3k) [inline scaladoc/resources/dotty_res/scripts/ux.js:288]: Let's preserve the state:
```suggestion
        history.replaceState(history.state, "", window.location.pathname + window.location.search);
```

**Comment by Dedelweiss** [comment]
> Thanks, that seems to be working fine. I left one suggestion.
> 
> I was wondering whether we could do something to not remove this part or URL, if user wants to link to specific section, but let's leave that for future work.

Thank you for your feedback @Florian3k, yes I was wondering too,indeed it can be useful. I can take a look at it for another PR.

File: scaladoc/resources/dotty_res/scripts/ux.js (hunk 1 above)
Comment (Dedelweiss) [reply scaladoc/resources/dotty_res/scripts/ux.js]: Done, thank you

**Review by Florian3k — APPROVED** [review]

**Pull request merged**